xds/src/main/java/io/grpc/xds/client/XdsClientImpl.java

```diff
@@ -196,22 +159,53 @@ public Map<String, XdsResourceType<?>> getSubscribedResourceTypesWithTypeUrl() {
     return Collections.unmodifiableMap(subscribedResourceTypeUrls);
   }
 
+  private ControlPlaneClient getActiveCpc(String authority) {
+    List<ControlPlaneClient> controlPlaneClients = activeCpClients.get(authority);
+    if (controlPlaneClients == null || controlPlaneClients.isEmpty()) {
+      return null;
+    }
+
+    return controlPlaneClients.get(controlPlaneClients.size() - 1);
+  }
+
+  @VisibleForTesting
+  public Object getActiveCpcForTest(String authority) {
+    return getActiveCpc(authority);
+  }
+
   @Nullable
   @Override
-  public Collection<String> getSubscribedResources(ServerInfo serverInfo,
-                                                   XdsResourceType<? extends ResourceUpdate> type) {
+  public Collection<String> getSubscribedResources(
```

Comment:
I went with the authority approach instead of the ServerInfo because fallback is per authority, so you can have a situation where one authority was complete and another did fallback and requests resources, the first authority's resources would include the fallback server's ServerInfo.  We don't want to get an update from the fallback server for the first authority so we don't want to request its resources.